cmd/service/init.go

```diff
@@ -52,41 +38,103 @@ mesg-core service init --current`,
 	DisableAutoGenTag: true,
 }
 
+func init() {
+	Init.Flags().StringP("name", "n", "", "Name")
+	Init.Flags().StringP("description", "d", "", "Description")
+	Init.Flags().BoolP("current", "c", false, "Create the service in the current path")
+	Init.Flags().StringP("template", "t", "", "Specify a template to download for your application")
+}
+
 func initHandler(cmd *cobra.Command, args []string) {
 	fmt.Printf("%s\n", aurora.Bold("Initialization of a new service"))
 
-	res := buildService(cmd)
-
-	mesgFile, err := generateMesgFile(&res)
+	tmpl := &template{
+		URL:  cmd.Flag("template").Value.String(),
+		Name: cmd.Flag("template").Value.String(),
+	}
+	if tmpl.URL == "" {
+		templates, err := getTemplates(templatesURL)
+		utils.HandleError(err)
+		tmpl, err = selectTemplate(templates)
+		utils.HandleError(err)
+		if tmpl == nil {
+			os.Exit(0)
+		}
+	}
+	path, err := downloadTemplate(tmpl)
+	utils.HandleError(err)
+	defer os.RemoveAll(path)
+	replacements := askReplacements(cmd)
+	folder := strings.Replace(strings.ToLower(replacements["name"]), " ", "-", -1)
+	if cmd.Flag("current").Value.String() == "true" {
+		folder = "./"
+	}
+	err = copyDir(path+"/template", folder, replacements)
 	utils.HandleError(err)
+	fmt.Println(aurora.Green("Service created in folder: " + folder))
+}
 
-	ok := false
-	fmt.Println()
-	fmt.Println(string(mesgFile))
-	if survey.AskOne(&survey.Confirm{Message: "Is this correct?", Default: true}, &ok, nil) != nil {
-		os.Exit(0)
+func getTemplates(url string) (templates []*template, err error) {
+	client := http.Client{}
+
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return
 	}
-	if !ok {
+	res, err := client.Do(req)
+	if err != nil {
+		return
+	}
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
 		return
 	}
 
-	folder := strings.Replace(strings.ToLower(res.Name), " ", "-", -1)
-	if cmd.Flag("current").Value.String() == "true" {
-		folder = "./"
+	err = json.Unmarshal(body, &templates)
+	return
+}
+
+func selectTemplate(templates []*template) (tmpl *template, err error) {
+	var result string
+	if survey.AskOne(&survey.Select{
+		Message: "Select your template",
+		Options: templatesToOption(templates),
+	}, &result, nil) != nil {
+		os.Exit(0)
 	}
-	err = writeInFolder(folder, mesgFile)
-	utils.HandleError(err)
-	fmt.Printf("%s\n", aurora.Green("Service created with success in folder '"+folder+"'").Bold())
+	tmpl = getTemplateResult(result, templates)
+	return
 }
 
-func generateMesgFile(service *service.Service) (res []byte, err error) {
-	var doc bytes.Buffer
-	tmpl, err := template.New("service-init").Parse(templateText)
+func templatesToOption(templates []*template) (options []string) {
+	options = []string{}
+	for _, template := range templates {
+		options = append(options, template.Name+" ("+template.URL+")")
+	}
+	options = append(options, addMyOwn)
```

Comment:
Could you also ask a "Enter template URL" entry? When the user select it, we then ask for the template URL. It will be the question version of the `--template` flag.

Reply:
The good thing with that and also the `--template` option is that you can use a local template, I just tested now and it's working so this is awesome for testing :)